batch/batch/worker.py

```diff
@@ -128,12 +138,15 @@ def docker_call_retry(timeout, name):
                 except DockerError as eget:
                     # 404 No such container
                     if eget.status == 404:
-                        error += 1
-                        if error < 10:
-                            delay = await sleep_and_backoff(delay)
+                        try_again = await handle_error()
+                        if try_again:
                             continue
-                        log.exception(f'encountered 10 errors while creating container {name}, aborting', stack_info=True)
                         raise
+            # No such image: gcr.io/...
+            if e.status == 404 and 'No such image' in e.message:
+                try_again = await handle_error()
+                if try_again:
+                    continue
             raise
```

Comment:
Rather than
```
try_again = await handle_error()
if try_again:
    continue
raise
```
I kind of prefer redefining handle_error:
```
async def handle_error(e):
    ...
        return
    log.exception(...)
    raise ValueError('too many failures in create_container') from e
```
and then:
```
await handle_error(e)
```

This is kind of where exceptions shine: returning through multiple layers of function calls.